WEB-INF/lib/auth/Auth_ldap.class.php

```diff
@@ -190,6 +189,10 @@ function authenticate($login, $password)
         }
       }
 
+      if (!empty($entries[0]['mail'][0])) {
```

Comment:
If present, use the first (primary) mail address of the user as login for further usage in the application.